### DISCUSSION TITLE
Remove test results from history

### DISCUSSION THREAD

**Discussion by bsamseth** [monkeytypegame/monkeytype#980] [Ideas] [answered]
I'd like the ability to delete test results from my history. 

When I first found the site I did a couple of tests on my laptop on my default layout. However, ever since I've been exclusively using monkeytype to practice a different layout, which I'm significantly slower with. The result is that I have some high scores that have been sitting untouched for 200+ tests, and likely won't be surpassed in a long while either. It is a small thing, but I'd really love to remove my highest scores so that I again can enjoy the motivation of hitting PBs.

I was able to partially solve it using tags, by tagging the highest scores with a qwerty-tag, and only showing non-tagged tests in the history graph. At least this way the progression looks right, but this doesn't change the high score.

Allowing this could potentially enable folks to boost their averages by removing tests that didn't go well (the opposite of my use-case), but I don't really see the harm in that, as it won't affect the leaderboards.

Is this something that would be welcome in monkeytype? I have little experience with web, but I do write code for a living so I'd be willing to make a PR if we end up deciding to go ahead with this, and spec out how the UI should be.

Love the site, by **far** my favorite typing site!

**Comment by Miodec** [comment, marked as answer]
I dont really like the idea of people removing tests because as you said - it would allow people to cherry pick results.

If you wanted to make a PR - how about a button in the `danger zone` section of the settings to reset your personal bests? You could do this by using the `SimplePopup` for the confirmation and create a new cloud function that clears the `personalBests` key in the user object.

**Reply by bsamseth** [reply]
Is the idea that by clearing the `personalBests` key any subsequent test would be marked as the new best score, regardless of any previous results? I suppose that would work, as long as the rest of the code doesn't make any assumptions about "results exist => a high score exists" that would crash if this was implemented.

I will see when/if time allows and see if I can make it work. Thanks anyway for a quick response!

**Comment by charlot-shaw** [comment]
I'd like this, because someone else sat down with a keyboard and beat my personal best. Props to them, but I'd like if the stats were all my own. Even if it's not a hard-delete, something to suppress a test from being graphed would be good.

**Reply by bsamseth** [reply]
The option to reset the personal bests was added way back when. Not that it sounds like what you'd want. 
![image](https://github.com/user-attachments/assets/62bf068e-b475-4581-933c-76df93c52635)
